source/adapter.screenshare.js

```diff
@@ -26,147 +55,284 @@ AdapterJS.defineMediaSourcePolyfill = function () {
+          });
+        }, 1);
+      // Regular getUserMedia() call
+      } else {
         baseGetUserMedia(constraints, successCb, failureCb);
       }
     };
 
     AdapterJS.getUserMedia = window.getUserMedia = navigator.getUserMedia;
-    /* Comment out to prevent recursive errors
-    navigator.mediaDevices.getUserMedia = function(constraints) {
+    // Comment out to prevent recursive errors as webrtc/adapter polyfills navigator.getUserMedia and calls
+    //   navigator.mediaDevices.getUserMedia internally
+    /*navigator.mediaDevices.getUserMedia = function(constraints) {
       return new Promise(function(resolve, reject) {
         window.getUserMedia(constraints, resolve, reject);
       });
     };*/
 
   } else if (window.navigator.webkitGetUserMedia && window.webrtcDetectedBrowser !== 'safari') {
     baseGetUserMedia = window.navigator.getUserMedia;
+    var iframe = document.createElement('iframe');
 
     navigator.getUserMedia = function (constraints, successCb, failureCb) {
-      if (constraints && constraints.video && !!constraints.video.mediaSource) {
-        if (window.webrtcDetectedBrowser !== 'chrome') {
-          // This is Opera, which does not support screensharing
+      // Append checks for overrides as these are mandatory
+      // Browsers (not Firefox since they went Promise based) does these checks and they can be quite useful
+      if (!(constraints && typeof constraints === 'object')) {
+        throw new Error('GetUserMedia: (constraints, .., ..) argument required');
+      } else if (typeof successCb !== 'function') {
+        throw new Error('GetUserMedia: (.., successCb, ..) argument required');
+      } else if (typeof failureCb !== 'function') {
+        throw new Error('GetUserMedia: (.., .., failureCb) argument required');
+      }
+
+      // Prevent accessing property from Boolean errors
+      if (constraints.video && typeof constraints.video === 'object' && constraints.video.hasOwnProperty('mediaSource')) {
+        var updatedConstraints = clone(constraints);
+        // See: https://developer.chrome.com/extensions/desktopCapture#type-DesktopCaptureSourceType
+        var mediaSourcesList = ['window', 'screen', 'tab', 'audio'];
+
+        // Check if it is Android phone for experimental 59 screensharing
+        // See: https://bugs.chromium.org/p/chromium/issues/detail?id=487935
+        if (navigator.userAgent.toLowerCase().indexOf('android') > -1) {
+          if (Array.isArray(updatedConstraints.video.mediaSource) ?
+            updatedConstraints.video.mediaSource.indexOf('screen') > -1 :
+            updatedConstraints.video.mediaSource === 'screen') {
+            updatedConstraints.video.mandatory = updatedConstraints.video.mandatory || {};
+            updatedConstraints.video.mandatory.chromeMediaSource = 'screen';
+            updatedConstraints.video.mandatory.maxHeight = window.screen.height;
+            updatedConstraints.video.mandatory.maxWidth = window.screen.width;
```

Comment:
Hope these height and width settings are correct. I wonder why it is needed only for android. not for desktop.